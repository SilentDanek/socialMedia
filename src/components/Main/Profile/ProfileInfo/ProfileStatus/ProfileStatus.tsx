import React, {useEffect, useState} from "react";

export function ProfileStatus({status, updateStatus}: any) {
    const [isStatusEditMode, setStatusEditMode] = useState(false);
    const [statusText, setStatusText] = useState(status as string);

    useEffect(() => setStatusText(status),[status])


    const toggleEditMode = () => {
        setStatusEditMode(!isStatusEditMode);
        if (isStatusEditMode && statusText !== status) {
            updateStatus(statusText);
        }
    }

    const onStatusChange = (e:React.ChangeEvent<HTMLInputElement>) => setStatusText(e.target.value);

    return (
        <div>
            <div>
                {isStatusEditMode ?
                    <div>
                        <input autoFocus
                               type="text"
                               onBlur={toggleEditMode}
                               onChange={onStatusChange}
                               value={statusText}/>
                    </div>:
                    <div>
                        <span onDoubleClick={toggleEditMode}>{status || "No status"}</span>
                    </div>
                }
            </div>
        </div>
    );
}