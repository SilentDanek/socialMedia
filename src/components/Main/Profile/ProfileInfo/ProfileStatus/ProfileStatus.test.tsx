import React from 'react';
import { render, screen, fireEvent} from '@testing-library/react';
import { ProfileStatus } from './ProfileStatus';

// Mock функции
const mockUpdateStatus = jest.fn();

describe('ProfileStatus component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('renders status from props', () => {
        render(<ProfileStatus status="Test status" updateStatus={mockUpdateStatus} />);
        const statusElement = screen.getByText("Test status");
        expect(statusElement).toBeInTheDocument();
    });

    test('renders "No status" when no status is provided', () => {
        render(<ProfileStatus status="" updateStatus={mockUpdateStatus} />);
        const statusElement = screen.getByText("No status");
        expect(statusElement).toBeInTheDocument();
    });

    test('enters edit mode on double click', () => {
        render(<ProfileStatus status="Test status" updateStatus={mockUpdateStatus} />);
        const statusElement = screen.getByText("Test status");
        fireEvent.doubleClick(statusElement);

        const inputElement = screen.getByDisplayValue("Test status");
        expect(inputElement).toBeInTheDocument();
    });

    test('updates status on input change', () => {
        render(<ProfileStatus status="Test status" updateStatus={mockUpdateStatus} />);
        fireEvent.doubleClick(screen.getByText("Test status"));

        const inputElement = screen.getByDisplayValue("Test status");
        fireEvent.change(inputElement, { target: { value: 'New status' } });

        expect(inputElement).toHaveValue('New status');
    });

    test('calls updateStatus with new value on blur', () => {
        render(<ProfileStatus status="Test status" updateStatus={mockUpdateStatus} />);
        fireEvent.doubleClick(screen.getByText("Test status"));

        const inputElement = screen.getByDisplayValue("Test status");
        fireEvent.change(inputElement, { target: { value: 'New status' } });
        fireEvent.blur(inputElement);

        expect(mockUpdateStatus).toHaveBeenCalledWith('New status');
    });

    test('does not call updateStatus if status is unchanged', () => {
        render(<ProfileStatus status="Test status" updateStatus={mockUpdateStatus} />);
        fireEvent.doubleClick(screen.getByText("Test status"));

        const inputElement = screen.getByDisplayValue("Test status");
        fireEvent.blur(inputElement);

        expect(mockUpdateStatus).not.toHaveBeenCalled();
    });

    test('matches the snapshot', () => {
        const { container } = render(<ProfileStatus status="Test status" updateStatus={() => {}} />);
        expect(container).toMatchSnapshot();
    });
});
