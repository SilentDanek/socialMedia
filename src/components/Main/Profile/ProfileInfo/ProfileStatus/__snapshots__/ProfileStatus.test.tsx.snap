// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus component matches the snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <span>
          Test status
        </span>
      </div>
    </div>
  </div>
</div>
`;
